contracts/BSCValidatorSet.sol

```diff
@@ -207,151 +228,160 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
+            emit failReasonWithStr("duplicate consensus address of validatorSet");
+            return ERROR_FAIL_CHECK_VALIDATORS;
+          }
+        }
+      }
     }
 
     // step 0: force all maintaining validators to exit `Temporary Maintenance`
     // - 1. validators exit maintenance
     // - 2. clear all maintainInfo
     // - 3. get unjailed validators from validatorSet
-    Validator[] memory validatorSetTemp = _forceMaintainingValidatorsExit(validatorSet);
-
-    //step 1: do calculate distribution, do not make it as an internal function for saving gas.
-    uint crossSize;
-    uint directSize;
-    for (uint i = 0;i<currentValidatorSet.length;i++) {
-      if (currentValidatorSet[i].incoming >= DUSTY_INCOMING) {
-        crossSize ++;
-      } else if (currentValidatorSet[i].incoming > 0) {
-        directSize ++;
-      }
-    }
-
-    //cross transfer
-    address[] memory crossAddrs = new address[](crossSize);
-    uint256[] memory crossAmounts = new uint256[](crossSize);
-    uint256[] memory crossIndexes = new uint256[](crossSize);
-    address[] memory crossRefundAddrs = new address[](crossSize);
-    uint256 crossTotal;
-    // direct transfer
-    address payable[] memory directAddrs = new address payable[](directSize);
-    uint256[] memory directAmounts = new uint256[](directSize);
-    crossSize = 0;
-    directSize = 0;
-    uint256 relayFee = ITokenHub(TOKEN_HUB_ADDR).getMiniRelayFee();
-    if (relayFee > DUSTY_INCOMING) {
-      emit failReasonWithStr("fee is larger than DUSTY_INCOMING");
-      return ERROR_RELAYFEE_TOO_LARGE;
-    }
-    for (uint i = 0;i<currentValidatorSet.length;i++) {
-      if (currentValidatorSet[i].incoming >= DUSTY_INCOMING) {
-        crossAddrs[crossSize] = currentValidatorSet[i].BBCFeeAddress;
-        uint256 value = currentValidatorSet[i].incoming - currentValidatorSet[i].incoming % PRECISION;
-        crossAmounts[crossSize] = value.sub(relayFee);
-        crossRefundAddrs[crossSize] = currentValidatorSet[i].BBCFeeAddress;
-        crossIndexes[crossSize] = i;
-        crossTotal = crossTotal.add(value);
-        crossSize ++;
-      } else if (currentValidatorSet[i].incoming > 0) {
-        directAddrs[directSize] = currentValidatorSet[i].feeAddress;
-        directAmounts[directSize] = currentValidatorSet[i].incoming;
-        directSize ++;
-      }
-    }
-
-    //step 2: do cross chain transfer
-    bool failCross = false;
-    if (crossTotal > 0) {
-      try ITokenHub(TOKEN_HUB_ADDR).batchTransferOutBNB{value:crossTotal}(crossAddrs, crossAmounts, crossRefundAddrs, uint64(block.timestamp + expireTimeSecondGap)) returns (bool success) {
-        if (success) {
-           emit batchTransfer(crossTotal);
-        } else {
-           emit batchTransferFailed(crossTotal, "batch transfer return false");
+    (Validator[] memory validatorSetTemp, ValidatorExtra[] memory validatorExtraSetTemp) = _forceMaintainingValidatorsExit(validatorSet, _validatorExtraSet);
```

Comment:
is it possible only return addressp[] too?

Reply:
Fixed